pkg/webhook/build.go

```diff
@@ -237,19 +237,13 @@ func validationError(reason, format string, fmtArgs ...interface{}) error {
 	}
 }
 
-func validateTimeout(timeout string) error {
+func validateTimeout(timeout metav1.Duration) error {
 	maxTimeout := time.Duration(24 * time.Hour)
 
-	if timeout == "" {
-		return nil
-	} else {
-		timeoutDuration, err := time.ParseDuration(timeout)
-		if err != nil {
-			return validationError("InvalidTimeFormat", "invalid build timeout %q err %#v. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout, err)
-		}
-		if timeoutDuration > maxTimeout {
-			return validationError("InvalidTimeout", "build timeout exceeded 24h")
-		}
+	if timeout.Duration > maxTimeout {
+		return validationError("InvalidTimeout", "build timeout exceeded 24h")
+	} else if timeout.Duration < 0 {
+		return validationError("InvalidTimeFormat", "invalid build timeout %q. Refer https://golang.org/pkg/time/#ParseDuration for time format documentation", timeout)
```

Comment:
This type should handle parse validation errors now, so this error message should just be "cannot define negative timeout" or something along those lines.